src/urllib3/request.py

```diff
@@ -37,26 +43,31 @@ class RequestMethods:
 
     _encode_url_methods = {"DELETE", "GET", "HEAD", "OPTIONS"}
 
-    def __init__(self, headers=None):
+    def __init__(self, headers: Optional[Mapping[str, str]] = None) -> None:
         self.headers = headers or {}
 
     def urlopen(
         self,
-        method,
-        url,
-        body=None,
-        headers=None,
-        encode_multipart=True,
-        multipart_boundary=None,
-        **kw,
+        method: str,
+        url: str,
+        body: Optional[HTTPBody] = None,
+        headers: Optional[Mapping[str, str]] = None,
+        encode_multipart: bool = True,
+        multipart_boundary: Optional[str] = None,
+        **kw: str,
     ) -> BaseHTTPResponse:  # Abstract
         raise NotImplementedError(
             "Classes extending RequestMethods must implement "
             "their own ``urlopen`` method."
         )
 
     def request(
-        self, method, url, fields=None, headers=None, **urlopen_kw
+        self,
+        method: str,
+        url: str,
+        fields: Optional[_TYPE_ENCODE_URL_FIELDS] = None,
```

Comment:
I'd rather allow the full range of fields here so users don't get type errors for passing in valid options for `request_encode_body`. Only wanted the `_TYPE_ENCODE_URL_FIELDS` to apply to `request_encode_url`.

Reply:
If I change it like:
`fields: Union[_TYPE_FIELDS, _TYPE_ENCODE_URL_FIELDS] = None,`

we still get this error:

```error: Argument "fields" to "request_encode_url" of "RequestMethods" has incompatible type "Union[Union[Sequence[Tuple[str, Union[str, bytes]]], Mapping[str, Union[str, bytes]], None], Union[Sequence[Union[Tuple[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]], RequestField]], Dict[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]]]]"; expected "Union[Sequence[Tuple[str, Union[str, bytes]]], Mapping[str, Union[str, bytes]], None]"```

It says that expexted type is `Union[Sequence[Tuple[str, Union[str, bytes]]], Mapping[str, Union[str, bytes]], None]` which is exists in type.

any idea?